File: rust/401.binary-watch.rs
/*
 * @lc app=leetcode id=401 lang=rust
 *
 * [401] Binary Watch
 */

// @lc code=start
impl Solution {
    pub fn read_binary_watch(turned_on: i32) -> Vec<String> {
        let turned_on = turned_on as u32;
        let mut times = Vec::new();
        if turned_on < 9 {
            for h in 0..12i32 {
                for m in 0..60i32 {
                    if h.count_ones() + m.count_ones() == turned_on {
                        times.push(format!("{}:{:02}", h, m));
                    }
                }
            }
        }
        return times;
    }
}
// @lc code=end

File: javascript/401.binary-watch.js
/*
 * @lc app=leetcode id=401 lang=javascript
 *
 * [401] Binary Watch
 */

// @lc code=start
/**
 * @param {number} turnedOn
 * @return {string[]}
 */
var readBinaryWatch = function(turnedOn) {
    function countOnes(num) {
      let ones = 0;
      while (num) {
        if (num & 1) ones++;
        num >>= 1;
      }
      return ones;
    }

    const times = [];
    if (turnedOn < 9) {
      for (let h = 0; h < 12; h++) {
        for (let m = 0; m < 60; m++) {
          if (countOnes(h) + countOnes(m) === turnedOn) times.push(`${h}:${m < 10 ? '0' : ''}${m}`)
        }
      }
    }
    return times;
};
// @lc code=end

